migrate from local storage to cookies for jwt token

adarshm11 opened the pull request:
addresses issue #1788 

now we don't have local storage of jwtToken, only cookies!!!

adarshm11 committed:
migrate from local storage to cookies for jwt token

joshua-demo commented on src/Components/Navbar/AdminNavbar.js:
```diff
@@ -13,9 +14,11 @@ export default function UserNavBar(props) {
   function handleLogout() {
     setAuthenticated(false);
+    cookies.remove('jwtToken');
     window.localStorage.removeItem('jwtToken');
```
Remove local storage line

joshua-demo commented on src/Components/Navbar/NavBarWrapper.js:
```diff
@@ -2,17 +2,20 @@ import React from 'react';
     setAuthenticated(false);
     setUser({});
+    cookies.remove('jwtToken');
     window.localStorage.removeItem('jwtToken');
```
Remove local storage line

joshua-demo commented on src/Pages/Login/Login.js:
```diff
@@ -1,21 +1,24 @@
     if (!loginStatus.error) {
       setAuthenticated(true);
-      window.localStorage.setItem('jwtToken', loginStatus.token);
+      cookies.set('jwtToken', loginStatus.token); // expire cookie after 1 week
```
There should be more settings we add here, like https only if we are in prod and same site secure

joshua-demo replied:
There should be more settings we add here, like https only if we are in prod and same site secure

adarshm11 committed:
move jwt logic to one file, add security features

charred-70 committed:
created button for led sign message expiration  (#1856)

* created button for expiration w/ no logic rn

* fixed conflict

* set expiration

* frontend core-v4 no longer fails with expiration button

* enabed led sign in config.json

* added sign_message class

* changed 'cancel' button to 'hide'

* made expiration button disappear when pressed and set expiration to null when updated without expiration

* polishing ui of expiration inputs

* made the expiration button ui look good on pc and mobile :D - char & tiffany

* added condition for expiration that is in the past

* i was in room 2 wbu

* made led sign page send expiration date in utc

* and thats the internship

* lint

* remove src/Pages/LedSign/sign_message.py

* undo button classname change

* separate variable

* why not await

* npx mocha test/api/LedSign.js

---------

Co-authored-by: Tyffoni <tyffoni@gmail.com>
Co-authored-by: evan <evanuxd@gmail.com>

iOliver678 committed:
Added door code to profile View (#1885)

* Added door code to profile View

* Update Profile.js

gursheyss committed:
change venmo to jonathan (#1887)

thebeninator committed:
printer now rejects expired sessions (#1891)

* printer now rejects expired sessions

* fix lint

adarshm11 committed:
fixed some misnamed html attributes (#1884)

woohoo

luuisabelle committed:
add metric to see which endpoint is not loaded through gauge values (#1906)

adarshm11 committed:
Correct decodeToken behavior (#1877)

* combine decodeTokenFromBodyOrQuery into decodeToken

* fixed misreference

* optimize logic for member access level

* fixed bug with decodeToken

* refactor decodeToken to handle unauthorized, forbidden, ok

* global use of decodeToken only

* tests pass, lint wont

* all green (?)

* remove commented out code, undo stray diff

* how safe are the user related apis

* specify access level for some apis

* const { membershipState } = require('../../../src/Enums.js');

---------

Co-authored-by: evan <evanuxd@gmail.com>

adarshm11 committed:
fix led sign and cleezy bug (#1927)

adarshm11 committed:
remove unused alumni role (#1928)

adarsh sux he typed that comment

adarshm11 committed:
slight index error (#1930)

DavidN016 committed:
OfficeAccessCard editing - backend changes (#1909)

* Add edit alias functionality for OfficeAccessCard (#1900)

- Implemented a new endpoint to edit card aliases, including validation for input.
- Updated the OfficeAccessCard utility to support alias editing.
- Enhanced the CardReader component to allow users to edit card aliases directly in the UI.
- Added corresponding tests to ensure proper functionality and error handling for the new feature.
- Introduced a new audit log action for alias edits.

* fixed lint issues

* fixed issues with edit alias

* fixed input re-sizing

* fix token decoding for office cards

* fix failing test

* separate frontend changes

* Add edit alias functionality for OfficeAccessCard (#1900)

- Implemented a new endpoint to edit card aliases, including validation for input.
- Updated the OfficeAccessCard utility to support alias editing.
- Enhanced the CardReader component to allow users to edit card aliases directly in the UI.
- Added corresponding tests to ensure proper functionality and error handling for the new feature.
- Introduced a new audit log action for alias edits.

* fixed lint issues

* fixed issues with edit alias

* fixed input re-sizing

* fix token decoding for office cards

* fix failing test

* separate frontend changes

* modify required checks for /edit

* addressed requested changes

* try to fix failing test

* try to fix that failing test one more time

* remove this test cuz it wont work

* OfficeAccessCard editing - UI changes (#1935)

* Add edit alias functionality for OfficeAccessCard (#1900)

- Implemented a new endpoint to edit card aliases, including validation for input.
- Updated the OfficeAccessCard utility to support alias editing.
- Enhanced the CardReader component to allow users to edit card aliases directly in the UI.
- Added corresponding tests to ensure proper functionality and error handling for the new feature.
- Introduced a new audit log action for alias edits.

* fixed lint issues

* fixed issues with edit alias

* fixed input re-sizing

* fix token decoding for office cards

* fix failing test

* card reader frontend

* rebase

* undo diff

* fixed requested changes

* fixed requested changes

---------

Co-authored-by: DavidN016 <david.a.nguyen01@sjsu.edu>

---------

Co-authored-by: DavidN016 <david.a.nguyen01@sjsu.edu>
Co-authored-by: adarshm11 <amlearning2023@gmail.com>
Co-authored-by: adarsh <110150037+adarshm11@users.noreply.github.com>

adarshm11 committed:
migrate from local storage to cookies for jwt token

adarshm11 committed:
move jwt logic to one file, add security features

adarshm11 committed:
Merge branch 'Adarsh/jwt-cookies' of https://github.com/SCE-Development/Clark into Adarsh/jwt-cookies
